datalogger changes

rumachan opened the pull request:
Do you have time to review this?

Added some new CR6 loggers to assests/dataloggers.csv
Moved some loggers around in install/dataloggers.csv. These were due to changes in the field yesterday (and in one case in January).

None of these sites have comms, so there are no downstream impacts.

elidana reviewed (commented):
Hi @rumachan ,

this looks good to me just looking at the changes. Just to do the "proper reviewer" job, is there any data I can look at to confirm timestamps? (happy to discuss that offline if easier)

rumachan commented:
Thank @elidana There are files,but if it's okay I'll paste you some file segments. These sites have no comms so nothing is available on-line.

This shows start of 04108 at RU001. I made end of previous logger 1 second earlier.
```
"TOA5","CR6 4108 RCL USB","CR6","4108","CR6.Std.11.01","CPU:RU001-ruapehucraterlake-v003.CR6","60071","Data_Table"
"TIMESTAMP","RECORD","LoggerName","temp_01_01","temp_01_01_Min","temp_01_01_Max","level_02_01","level_02_01_Min","level_02_01_Max"
"TS","RN","","","","","","",""
"","","Smp","Smp","Min","Max","Smp","Min","Max"
"2022-01-11 20:30:00",0,"RU001","NAN","NAN","NAN","NAN","NAN","NAN"
"2022-01-17 22:40:00",1,"RU001",30.18,27.74,30.18,0.756,0.753,0.919
"2022-01-17 22:50:00",2,"RU001",35.38,30.42,35.56,0.751,0.746,0.754
"2022-01-17 23:00:00",3,"RU001",33.84,33.84,34.18,0.754,0.751,0.754
"2022-01-18 00:00:00",4,"RU001",34.04,33.8,34.04,0.753,0.753,0.755
"2022-01-18 00:10:00",5,"RU001",34.39,34.08,34.39,0.754,0.753,0.754
"2022-01-18 00:20:00",6,"RU001",34.46,34.4,34.47,0.753,0.753,0.755
"2022-01-18 00:30:00",7,"RU001",34.3,34.3,34.45,0.753,0.753,0.754
```
TOUTL logger change
File from old logger
```
"TOA5","CR6 4117 UpperTamaLake USB","CR6","4117","CR6.Std.04","CPU:TOUTL-uppertamalake-v005.CR6","8497","Data_Table"
"TIMESTAMP","RECORD","LoggerName","level_01_01","level_01_01_Min","level_01_01_Max","temp_01_02","temp_01_02_Min","temp_01_02_Max","temp_02_01","temp_02_01_Min","temp_02_01_Max"
"TS","RN","","","","","","","","","",""
"","","Smp","Smp","Min","Max","Smp","Min","Max","Smp","Min","Max"
"2022-01-17 21:00:00",0,"TOUTL",2.284,2.284,2.287,18.36,18.33,18.36,17.96,17.96,17.96
"2022-01-17 21:10:00",1,"TOUTL",2.285,2.284,2.286,18.37,18.36,18.37,17.96,17.96,17.96
"2022-01-17 21:20:00",2,"TOUTL",2.284,2.284,2.289,18.41,18.38,18.41,18,17.96,18
later part of file
"2022-03-30 20:00:00",10361,"TOUTL",1.532,1.531,1.534,12.77,12.75,12.77,12.6,12.57,12.6
"2022-03-30 20:10:00",10362,"TOUTL",1.532,1.531,1.533,12.8,12.78,12.8,12.62,12.6,12.62
"2022-03-30 20:20:00",10363,"TOUTL",1.532,1.53,1.534,12.83,12.8,12.83,12.66,12.63,12.66
"2022-03-30 20:30:00",10364,"TOUTL",1.53,1.53,1.533,12.87,12.83,12.87,12.7,12.66,12.7
```
File from new logger
```
"TOA5","CR6 17444 UpperTamaLake USB","CR6","17444","CR6.Std.11.01","CPU:TOUTL-uppertamalake-v005.CR6","8497","Data_Table"
"TIMESTAMP","RECORD","LoggerName","level_01_01","level_01_01_Min","level_01_01_Max","temp_01_02","temp_01_02_Min","temp_01_02_Max","temp_02_01","temp_02_01_Min","temp_02_01_Max"
"TS","RN","","","","","","","","","",""
"","","Smp","Smp","Min","Max","Smp","Min","Max","Smp","Min","Max"
"2022-03-30 20:40:00",1,"TOUTL",1.532,1.531,1.533,12.91,12.89,12.91,12.74,12.72,12.74
```

TO063 logger change

File from old logger
```
"TOA5","CR6 15369 LowerEmeraldSpring USB","CR6","15369","CR6.Std.10.02","CPU:TO063-loweremeraldspring-v002.CR6","60184","Data_Table"
"TIMESTAMP","RECORD","LoggerName","temp_01_01","temp_01_01_Min","temp_01_01_Max","temp_02_01","temp_02_01_Min","temp_02_01_Max","temp_03_01","temp_03_01_Min","temp_03_01_Max"
"TS","RN","","","","","","","","","",""
"","","Smp","Smp","Min","Max","Smp","Min","Max","Smp","Min","Max"
"2021-10-14 23:10:00",0,"TO063",3.297,0.282,3.297,62.88,60.65,63.02,39.05,32.22,39.05
"2021-10-14 23:20:00",1,"TO063",6.12,4.084,6.12,62.3,60.31,62.3,40.1,39.42,49.27
"2021-10-14 23:30:00",2,"TO063",8.55,6.356,8.55,64.5,62.58,64.5,52.41,41.01,52.41
later content
"2022-03-30 20:40:00",24030,"TO063",10.38,10.37,10.44,53.7,53.7,53.72,49.41,49.36,49.41
"2022-03-30 20:50:00",24031,"TO063",10.28,10.26,10.43,53.47,53.47,53.7,49.44,49.41,49.44
"2022-03-30 21:00:00",24032,"TO063",11.36,10.31,11.55,53.85,53.56,53.85,49.59,49.34,49.6
```

File from new logger
```
"TOA5","CR6 4115 LowerEmerald USB","CR6","4115","CR6.Std.11.01","CPU:TO063-loweremeraldspring-v002.CR6","60184","Data_Table"
"TIMESTAMP","RECORD","LoggerName","temp_01_01","temp_01_01_Min","temp_01_01_Max","temp_02_01","temp_02_01_Min","temp_02_01_Max","temp_03_01","temp_03_01_Min","temp_03_01_Max"
"TS","RN","","","","","","","","","",""
"","","Smp","Smp","Min","Max","Smp","Min","Max","Smp","Min","Max"
"2022-03-30 21:10:00",0,"TO063",13.19,13.19,15.48,55.37,55.37,57.56,51.16,51.16,53.15
"2022-03-30 21:20:00",1,"TO063",11.47,11.47,12.75,54.9,54.9,55.18,50.79,50.45,51.17
```

elidana commented on install/dataloggers.csv:
```diff
@@ -1,17 +1,20 @@
+Campbell Scientific,CR6,04115,Tongariro Lower Emerald Spring,,2022-03-30T21:10:00Z,9999-01-01T00:00:00Z
 Campbell Scientific,CR6,04116,Waimangu Inferno Crater,,2017-11-07T00:00:00Z,9999-01-01T00:00:00Z
-Campbell Scientific,CR6,04117,Upper Tama Lake,,2019-03-12T03:30:00Z,9999-01-01T00:00:00Z
+Campbell Scientific,CR6,04117,Upper Tama Lake,,2019-03-12T03:30:00Z,2022-03-30T19:39:59Z
```
Hi @rumachan , thanks a lot for sharing the data, really helpful!
should this closing time be some time after the last data point received from the old logger?
`"2022-03-30 20:30:00",10364,"TOUTL",1.53,1.53,1.533,12.87,12.83,12.87,12.7,12.66,12.7`

so with `2022-03-30T20:40:00Z` as starting date of new data logger, end date here to be
`2022-03-30T20:39:59Z` (or playing with minutes/seconds between 20:30 and 20:40 in case we have data gaps in between?)

rumachan replied:
That's a good question. The times are inclusive aren't they? So as long as there is no overlap and no data points fall outside the installed periods we are okay? I don't have the times of exactly when a logger was disconnected as it's not recorded. Just setting date-time to fit the data.

Samples are collected at 10 minute intervals so there can't be a data gap between 20:30 and 20:40, just data values at 20:30 and 20:40.

elidana replied:
yes, times are inclusive. The "approach" I was using for gnss was to have a start/stop timestamp during the data gap, to ensure that we were covering all available data with associated metadata. 
(and btw: good on the field technicians that can swap a datalogger within a 10 minutes interval...that's really impressive!)

and to summarise, the main change I was suggesting here was to: 
```diff
-Campbell Scientific,CR6,04117,Upper Tama Lake,,2019-03-12T03:30:00Z,2022-03-30T19:39:59Z
+Campbell Scientific,CR6,04117,Upper Tama Lake,,2019-03-12T03:30:00Z,2022-03-30T20:39:59Z
```

if you agree that the 04117 logger was removed sometime between 20:30 and 20:40

rumachan replied:
So I made a mistake with the end time? Hour should be 20, but I had 19? Now I get it

elidana replied:
he he yes, sorry my bad english! yes, that's what I meant (but I was also asking question about the data and swap procedure at the same time :stuck_out_tongue:

elidana reviewed (commented):
thanks a lot @rumachan for sharing the files!! helps a lot

rumachan commented:
I changed the TOUTL hour from 19 to 20. I got confused with git as it wouldn't let me push the changes so I changed the file in the repo on github.

Can you check again please.

elidana reviewed (approved):
looks good, and thanks a lot for updating the timestamp!
